crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs

```diff
@@ -833,6 +833,20 @@ impl KitsuneP2pHandler for Space {
         initial_arq: Option<Arq>,
     ) -> KitsuneP2pHandlerResult<()> {
         tracing::debug!(?space, ?agent, ?initial_arq, "handle_join");
+
+        match initial_arq {
+            // This agent has been on the network before and has an arc stored on the Kitsune host.
+            // Respect the current storage settings and apply this arc.
+            Some(initial_arq) => {
+                self.agent_arqs.insert(agent.clone(), initial_arq);
+            }
+            // This agent is new to the network and has no arc stored on the Kitsune host.
+            // Get a default arc for the agent
+            None => {
+                self.agent_arqs.insert(agent.clone(), self.default_arc_for_agent(agent.clone()));
+            }
+        }
```

Comment:
Don't you need to delete the old insert below?

Reply:
That's weird, I don't see anything there :S 

Thank you, yes, that can go. Won't actually do anything other than a duplicate insert but I didn't mean to leave it behind.